src/app/shared/data/allocations.service.ts

```diff
@@ -33,16 +33,37 @@ export class AllocationsService {
     return this.allocationsSubject$.getValue();
   }
 
-  addStudentToProject(studentId: string, projectId: string): void {
+  get allocations$(): Observable<Allocation[]> {
+    return this.allocationsSubject$.asObservable();
+  }
+
+  moveStudentToProject(studentId: string, projectId: string): void {
     this.removeStudentFromProjects(studentId);
 
     const allocations = this.getAllocations();
     const allocation = allocations.find(a => a.projectId === projectId);
```

Comment:
Something I overlooked previously (or wasn't part of your code): Could you please change 'a' to 'allocation'?